ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/PaywallDialog.kt

```diff
@@ -0,0 +1,80 @@
+package com.revenuecat.purchases.ui.revenuecatui
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Scaffold
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.ReadOnlyComposable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.window.Dialog
+import androidx.compose.ui.window.DialogProperties
+import androidx.window.core.layout.WindowWidthSizeClass
+import com.revenuecat.purchases.Purchases
+import com.revenuecat.purchases.PurchasesException
+import com.revenuecat.purchases.awaitCustomerInfo
+import com.revenuecat.purchases.ui.revenuecatui.helpers.Logger
+import com.revenuecat.purchases.ui.revenuecatui.helpers.computeWindowWidthSizeClass
+import kotlinx.coroutines.launch
+
+@Composable
+fun PaywallDialog(
+    paywallDialogOptions: PaywallDialogOptions,
+) {
+    val shouldDisplayBlock = paywallDialogOptions.shouldDisplayBlock
+    var shouldDisplayDialog by remember { mutableStateOf(shouldDisplayBlock == null) }
+    if (shouldDisplayBlock != null) {
+        LaunchedEffect(paywallDialogOptions) {
+            launch {
+                shouldDisplayDialog = try {
+                    // TODO-PAYWALLS: This won't receive updates in case the customer info changes and starts/stops
+                    // being eligible to display the paywall dialog. We would need to support multiple customer info
+                    // listeners to refresh this.
+                    val customerInfo = Purchases.sharedInstance.awaitCustomerInfo()
```

Comment:
I didn't fully like this here so I was thinking about moving it to the view model. But then, I didn't want to have to create the view model at this point (since we might not even display the dialog)... I thought it was ok for now to keep this. Lmk if you would prefer me to move it.

Reply:
This is fine. That's exactly how `PresentingPaywallModifier` works.